autoscale/src/main/java/com/sequenceiq/periscope/controller/validation/AlertValidator.java

```diff
@@ -90,13 +97,26 @@ public void validateDistroXAutoscaleClusterRequest(Cluster cluster,
 
         if (!distroXAutoscaleClusterRequest.getLoadAlertRequests().isEmpty()) {
             Set<String> loadAlertHostGroups = distroXAutoscaleClusterRequest.getLoadAlertRequests().stream()
+                    .map(loadAlertRequest -> {
+                        validateClusterLimit(loadAlertRequest.getScalingPolicy().getAdjustmentType(),
+                                loadAlertRequest.getLoadAlertConfiguration().getMaxResourceValue());
+                        return loadAlertRequest;
+                    })
                     .map(LoadAlertRequest::getScalingPolicy)
                     .map(ScalingPolicyBase::getHostGroup)
                     .collect(Collectors.toSet());
             validateSupportedHostGroup(cluster, loadAlertHostGroups, AlertType.LOAD);
         }
     }
 
+    public void validateClusterLimit(AdjustmentType adjustmentType, Integer configuredMax) {
+        if ((adjustmentType == AdjustmentType.LOAD_BASED || adjustmentType == AdjustmentType.EXACT)
+                && configuredMax > limitsConfigurationService.getMaxNodeCountLimit()) {
+            throw new BadRequestException(messagesService.getMessage(MessageCode.AUTOSCALING_CLUSTER_LIMIT_EXCEEDED,
+                    List.of(limitsConfigurationService.getMaxNodeCountLimit())));
```

Comment:
This check will be at the hostGroup level, correct? The limit on the cluster size spans across all hostGroups in Cloudbreak? - so this would allow the cluster to be configured to a size which it cannot actually scale out to?

Should the existing cluster size be checked before configuring this property? (and factor in the min/max that is being configured on the hostGroup being configured, and the current size of that hostGroup?)

Even after this, it is possible that a different hostGroup is increased in size - in which case - the cluster will not actually be able to reach the configured max capacity (if another hostGroup is grown). If such a scenario is hit during actual scaling - I think it will be very useful to include this in the message that shows up on the console. Something like
"Cannot scale up to X nodes, Will exceed maximum cluster node count. Instead scaling to ${Actual Node Count}" (Fix text to be more aligned with the current scaling message)

Reply:
CB does not support hostGroup level max (only total cluster limit) but for autoscaling we have to define hostgroup level max, hence this validation. 
But during actual scaling action, scale-up node count will be adjusted  based on actual cluster size at time of trigger.
Also added message to console indicating the same.